backend/spec/requests/api/v1/email_confirmation_spec.rb

```diff
@@ -80,37 +80,43 @@
       end
 
       response "422", "Invalid Token" do
-        let(:confirmation_token) { "invalid_token" }
+        context "when confirmation token is invalid" do
+          let(:confirmation_token) { "invalid_token" }
 
-        run_test!
+          run_test!
 
-        it "returns correct error", generate_swagger_example: true do
-          expect(response_json["errors"][0]["title"]).to eq("Confirmation token is invalid")
-          expect(session["warden.user.user.key"]).to be_nil
-          expect(unconfirmed_user.reload.confirmed?).to eq(false)
+          it "returns correct error", generate_swagger_example: true do
+            expect(response_json["errors"][0]["title"]).to eq("Confirmation token is invalid")
+            expect(session["warden.user.user.key"]).to be_nil
+            expect(unconfirmed_user.reload.confirmed?).to eq(false)
+          end
         end
 
         context "already confirmed" do
-          let(:confirmation_token) do
-            unconfirmed_user.confirm # not sure why it does not work in before block
-            unconfirmed_user.confirmation_token
-          end
+          let(:confirmation_token) { unconfirmed_user.confirmation_token }
 
-          it "returns correct error", generate_swagger_example: true do
+          before { unconfirmed_user.confirm }
```

Comment:
and now it works?

Reply:
So, we should always, run_test in separate contexts?

Reply:
or just only when we have problems with `before`.

Reply:
Yeah, but going back to previous version and adding `before` and things not working ;] haha